Add contentTextStyle support to SimpleDialog

777genius opened the pull request:
## Description

This PR adds `contentTextStyle` support to `SimpleDialog`, matching the functionality already available in `AlertDialog`.

### Problem

`SimpleDialog` was ignoring `DialogTheme.contentTextStyle`, even though the `DialogThemeData.contentTextStyle` documentation explicitly states it applies to `SimpleDialog.children`:

> "Overrides the default value for DefaultTextStyle for SimpleDialog.children and AlertDialog.content."

However, only `AlertDialog` actually implemented this behavior, forcing developers to specify text styles individually on each child widget.

### Solution

Added `contentTextStyle` parameter to `SimpleDialog` and applied it using `DefaultTextStyle`, following the same pattern as `AlertDialog`. The implementation uses the standard fallback chain:

`contentTextStyle` parameter → `DialogTheme.contentTextStyle` → M2/M3 defaults

### Testing

Added a new test `'SimpleDialog Custom Content Text Style'` that verifies `contentTextStyle` is correctly applied to the children of `SimpleDialog`.

## Related Issues

Fixes https://github.com/flutter/flutter/issues/59462

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

777genius committed:
Add contentTextStyle support to SimpleDialog

SimpleDialog now respects contentTextStyle from both the constructor
parameter and DialogTheme, matching AlertDialog behavior.

This implementation follows the existing pattern used for AlertDialog
and titleTextStyle. The DialogThemeData.contentTextStyle property was
already documented to apply to SimpleDialog.children, but was not
actually implemented.

Fixes #59462

dkwingsmt reviewed (approved):
LGTM. Thanks for your work! Also sorry for late review.

QuncCccccc commented on packages/flutter/lib/src/material/dialog.dart:
```diff
@@ -1288,7 +1301,7 @@ class SimpleDialog extends StatelessWidget {
     // The paddingScaleFactor is used to adjust the padding of Dialog
     // children.
     final TextStyle defaultTextStyle =
```
nit: I think we can rename the variable to effectiveTextStyle. Just feel defaultTextStyle is a bit confusing and effectiveXXX is convention we use for this path.

QuncCccccc commented on packages/flutter/test/material/dialog_test.dart:
```diff
@@ -303,6 +303,22 @@ void main() {
     expect(materialWidget.clipBehavior, Clip.antiAlias);
   });
 
+  testWidgets('SimpleDialog Custom Content Text Style', (WidgetTester tester) async {
```
Can we add one more unit test to berify DialogTheme.contentTextStyle works?

QuncCccccc reviewed (approved):
Thanks for adding this! LGTM. I'll add a g3fix to fix the Google testing. The failures look expected since now we added the missing default value for content text style (`defaults.contentTextStyle`).

777genius committed:
Address review feedback: rename variable and add tests

- Rename defaultTextStyle to effectiveTitleTextStyle to follow Flutter
  convention for resolved values (consistent with effectiveTitlePadding,
  effectiveContentPadding naming pattern)
- Add test to verify DialogTheme.contentTextStyle works with SimpleDialog
- Add Material3/Material2 theme fallback tests for contentTextStyle
  to ensure documented behavior is covered

777genius commented on packages/flutter/test/material/dialog_test.dart:
```diff
@@ -295,6 +295,71 @@ void main() {
+    expect(content.text.style, contentTextStyle);
+  });
+
+  testWidgets('SimpleDialog Custom Content Text Style - DialogTheme', (WidgetTester tester) async {
```
Done! Linux analyze passes now. The issue was that the branch was based on an older master.

777genius replied:
Thank you for review!

QuncCccccc committed:
Merge branch 'master' into fix-simpledialog-contenttextstyle-59462

QuncCccccc committed:
Merge branch 'master' into fix-simpledialog-contenttextstyle-59462
